quipucords/scanner/network/runner/roles/redhat_release/tasks/main.yml

```
@@ -4,25 +4,17 @@
   set_fact:
     internal_host_started_processing_role: "redhat_release"
 
-- name: initialize redhat_release facts
-  set_fact:
-    redhat_release_name: ''
-    redhat_release_version: ''
-    redhat_release_release: ''
-  ignore_errors: yes
-
 - name: gather internal_redhat_release fact
   raw: rpm -q --queryformat "%{NAME}\n%{VERSION}\n%{RELEASE}\n" --whatprovides redhat-release
   register: internal_redhat_release
-  ignore_errors: yes
+  ignore_errors: true
   when: 'internal_have_rpm'
 
 - name: set redhat_release facts based on internal_redhat_release
   set_fact:
     redhat_release_name: "{{ internal_redhat_release['stdout_lines'][0] }}"
     redhat_release_version: "{{ internal_redhat_release['stdout_lines'][1] }}"
     redhat_release_release: "{{ internal_redhat_release['stdout_lines'][2] }}"
-  ignore_errors: yes
+  ignore_errors: true
   when:
     - "'stdout_lines' in internal_redhat_release"
```

@nicolearagao do you know if ansible `when` short circuits? If so I think we can still remove the initial setup and add this

```suggestion
    - internal_redhat_release is defined
    - "'stdout_lines' in internal_redhat_release"
    - "internal_redhat_release['stdout_lines']|length == 3"
```

PS: I shouldn't have removed the length check. probably did this by mistake

---

It's not short circuit evaluation. 
Why don't you do something like this? 
when: result.stdout | default('') == 'foo'

---

Not sure if I got it. Would this work if `internal_redhat_release` is `undefined`?

```suggestion
    - internal_redhat_release.stdout | default("\n\n\n")
```

---

I apologize, my suggestion wasn't clear.  if internal_redhat_release is undefined, it means that either this is not a red hat machine or that there was a problem while collecting this information. 
For the further I think it's still worth it to have a default value if anything goes wrong. I don't think calling set_fact twice to do it is the most efficient way. 
You could do:
set_fact:
    redhat_release_name: "{{ internal_redhat_release['stdout_lines'][0] | default(" ") }}"
    redhat_release_version: "{{ internal_redhat_release['stdout_lines'][1] | default(" ")}}"
    redhat_release_release: "{{ internal_redhat_release['stdout_lines'][2] | default(" ")}}"
The reason why this won't blow up if internal_redhat_release is undefined is due to the flag ignore_errors. In any other scenario it would.

---

Setting a fact to " " can be problematic because the expected value for "empty" facts is "".
I could apply this suggestion with default to "" but I'm not confident this won't cause errors - it depends on 2 premises that might not be fulfilled (stdout_lines exists and has at least 3 lines).

Now I understood why this was implemented the way it was and I think I prefer this approach now. What do you think?